lighthouse-core/gather/computed/computed-artifact.js

```diff
@@ -38,18 +38,21 @@ class ComputedArtifact {
 
   /**
    * Request a computed artifact, caching the result on the input artifact.
-   * @param {!OBject} artifact
+   * @param {!Object} artifact
+   * @param {!Object=} artifacts
    * @return {!Promise}
    */
-  request(artifact) {
+  request(artifact, artifacts) {
```

Comment:
> Does this proposal allow for passing more than one artifact?

No, but we definitely need that too. I can add it in a followup PR (or you can if you want). Basically computedArtifact's `_cache` needs to support a tuple as a key instead of just objects directly. There's no really efficient way to do that in JS today, but luckily these caches all have like one item in them :)